### PR TITLE
Price per NFT & User Access

### PR DESCRIPTION
**Pull request by JaPilgrim**
Testing of Features & functionalities of #110 & #113 succeeds
I created & verified the source code, just commented out the "constructor". If now deployed with the "constructor", it gives the following messages and prohibits a re-verification for the contract WITH the constructor. 
image
Therefore I assume it to be verifiable.

### CODE REVIEW

File: blockchain/contracts/NFTtheWorld.sol [hunk 1]
```diff
@@ -3,23 +3,31 @@
...
-        address owner;
+        address payable owner;
         uint256 nftId;
         uint256 dropId;
         uint256 dropTime;
+        uint256 weiPrice;
         address reservedFor;
     }
 
+    mapping(address => bool) public adminRights;
```
Comment (dominic-heil) [inline blockchain/contracts/NFTtheWorld.sol:29]: Does this need to be public? I don't think it destroys anything, just curious.

File: blockchain/contracts/NFTtheWorld.sol [hunk 2]
```diff
@@ -3,23 +3,31 @@
 
 pragma solidity ^0.8.0;
 
+//import "./AddrArrayLib.sol";
+
+
 contract NFTtheWorld {
     address payable nftOwner;
 
+    address payable[] adminAddresses;
```
Comment (dominic-heil) [inline blockchain/contracts/NFTtheWorld.sol:12]: This variable was created but is never used?

**Review by dominic-heil — COMMENTED** [review]
Code looks fine. I tried to run the adminRights function on the blockchain (https://kovan.etherscan.io/address/0x2728c74325b2c3280cabc662dc3b05c906b8ae17#readContract) with my address and am not an admin as expected. I also tried to run the adminRights function with the contract creator address (0xe016160abedf3537bc3d7ba946a4dba3ee49ceed) but that address also was no admin. So I cannot confirm if the code works as intended.

File: blockchain/contracts/NFTtheWorld.sol (hunk 1 above)
Comment (JaPilgrim) [reply blockchain/contracts/NFTtheWorld.sol:29]: Hmm no not in general, but for right now & testing purposes it suits good I think

File: blockchain/contracts/NFTtheWorld.sol [hunk 3]
```diff
@@ -172,6 +193,23 @@ contract NFTtheWorld {
...
+
+    function addToAdmins(address payable _addressToAdd) public onlyByAdmins() {
+        require(!adminRights[msg.sender]=true, "Address already has Admin rights.");
+        adminRights[_addressToAdd]=true;      
+    }
+
+    function removeFromAdmins(address payable _addressToRemove) public onlyByAdmins() {
+        require(!adminRights[msg.sender]=false, "Address already has no Admin rights.");
+        adminRights[_addressToAdd]=false;  
+    }
```
[the remark this answers is not in the document]
Comment (JaPilgrim) [reply blockchain/contracts/NFTtheWorld.sol:210]: Good idea in general. I added it, but for testing I commented it out.

**Comment by JaPilgrim** [comment]
PR closed & code integrated into branch #120

**Pull request closed without merging**